sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java

```diff
@@ -123,16 +123,11 @@ public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineN
         return authorizeRequest(context).then(Mono.defer(next::process)).flatMap(httpResponse -> {
             String authHeader = httpResponse.getHeaderValue(HttpHeaderName.WWW_AUTHENTICATE);
             if (httpResponse.getStatusCode() == 401 && authHeader != null) {
-                return authorizeRequestOnChallenge(context, httpResponse).flatMap(retry -> {
-                    if (retry) {
-                        // Both Netty and OkHttp expect the requestBody to be closed after the response has been read.
-                        // Failure to do so results in memory leak.
-                        // In case of StreamResponse (or other scenarios where we do not eagerly read the response)
-                        // the response body may not be consumed.
-                        // This can cause potential leaks in the scenarios like above, where the policy
-                        // may intercept the response and it may never be read.
-                        // Forcing the read here - so that the memory can be released.
-                        return httpResponse.getBody().ignoreElements().then(nextPolicy.process());
```

Comment:
closing is better than reading to the end in edge cases:
- we don't need to waste time on reading (unlikely) big response
- if there is a transient issue during stream reading, we don't need to handle/retry it.